docs/docs/development/hardware-metadata-files.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Hardware Metadata Files
+---
+
+## Overview
+
+ZMK leverages an additional metadata YAML file for all boards and shields to provide high level information about the hardware to be leveraged by setup scripts/utilities, proper inclusion in the website hardware list, etc.
```

Comment:
Super minor: using "leverage" twice makes this read a bit awkwardly to me. Maybe just use "use" and/or "to be read by setup scripts" to keep things easy to understand?

(If you change this, Ctrl+F "leverage" since it's used in a few other places too.)